smac/tae/execute_ta_run.py

```diff
@@ -35,6 +35,18 @@ def enum_hook(obj):
                 return getattr(globals()[name], member)
         return obj
 
+class BudgetExhaustedException(Exception):
+    """ Exception indicating that time- or memory-budgets are exhausted. """
+    pass
+
+class TAEAbortException(Exception):
+    """ Exception indicating that the TAE suggests an ABORT of SMAC. """
+    pass
+
+class TAEFirstRunCrashedException(TAEAbortException):
```

Comment:
Hm, could we rename this to `FirstRunCrashedException`? I think this is easier for a user.